src/OpenTelemetry.Exporter.InMemory/InMemoryExporter.cs

```diff
@@ -0,0 +1,43 @@
+// <copyright file="InMemoryExporter.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using OpenTelemetry;
+
+namespace OpenTelemetry.Exporter
+{
+    public class InMemoryExporter<T> : BaseExporter<T>
+        where T : class
+    {
+        private readonly ICollection<object> exportedItems;
+
+        public InMemoryExporter(InMemoryExporterOptions options = null)
+        {
+            this.exportedItems = options?.ExportedItems;
+        }
+
+        public override ExportResult Export(in Batch<T> batch)
+        {
+            foreach (var data in batch)
```

Comment:
Prob faster to do the null check outside of the loop:
```csharp
if (this.exportedItems != null)
{
   foreach (var data in batch)
   {
      this.exportedItems.Add(data);
   }
}
```

Reply:
Nice catch.

I've changed it to return `Export.Failure` when `exportedItems` is null.